editor/test/editor/build_target_test.clj

```diff
@@ -181,21 +181,16 @@
         (g/error-fatal (.getMessage error))))))
 
 (defn- all-build-targets [project]
-  (transduce (comp (map second)
-                   (map build-targets-or-error)
-                   (remove g/error?)
-                   (remove empty?))
-             (completing
-               (fn [all-build-targets resource-build-targets]
-                 (let [seen-content-hashes (into #{}
-                                                 (map :content-hash)
-                                                 all-build-targets)]
-                   (into all-build-targets
-                         (pipeline/flatten-build-targets
-                           resource-build-targets seen-content-hashes)))))
-             []
-             (sort-by key
-                      (g/node-value project :nodes-by-resource-path))))
+  (build/resolve-dependencies
+    (->> (g/node-value project :nodes-by-resource-path)
+         (sort-by key)
+         (->Eduction
+           (comp
+             (map val)
+             (map build-targets-or-error)
+             (remove g/error?)
+             (remove empty?))))
```

Comment:
Isn't this just `eduction`? I feel the readability trade-off might not be worth it in some cases.

Reply:
That's fair :)